File: v1/src/main/java/com/api/v1/controllers/payment/PaymentRegistrationController.java
package com.api.v1.controllers.payment;

import com.api.v1.dtos.payment.PaymentResponseDto;
import com.api.v1.services.payment.PaymentRegistrationService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/payments")
public class PaymentRegistrationController {

    @Autowired
    private PaymentRegistrationService service;

    @PostMapping("{ssn}/{vin}/{cardNumber}")
    @ResponseStatus(value = HttpStatus.CREATED)
    public Mono<PaymentResponseDto> register(
            @PathVariable String ssn,
            @PathVariable String vin,
            @PathVariable String cardNumber
    ) {
        return service.register(ssn, vin, cardNumber);
    }

}

File: v2/src/main/kotlin/com/api/v2/payment/controllers/PaymentRegistrationController.kt
package com.api.v2.payment.controllers

import com.api.v2.payment.dtos.PaymentResponseDto
import com.api.v2.payment.services.PaymentRegistrationService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/v2/payments")
class PaymentRegistrationController {

    @Autowired
    private lateinit var service: PaymentRegistrationService

    @PostMapping("{ssn}/{vin}/{cardNumber}")
    @ResponseStatus(value = HttpStatus.CREATED)
    suspend fun register(
        @PathVariable ssn: String,
        @PathVariable vin: String,
        @PathVariable cardNumber: String
    ): PaymentResponseDto {
        return service.register(ssn, vin, cardNumber)
    }


}